pysignalscope/scope.py

```diff
@@ -732,6 +809,13 @@ def short_to_period(self, f0: Optional[float] = None, time_period: Optional[floa
         :param start_time: start time in seconds
         :type start_time: float
         """
+        if not isinstance(f0, (float, int)) != f0 is not None:
+            raise TypeError("f0 must be type float/int/None.")
+        if not isinstance(time_period, (float, int)) != f0 is not None:
+            raise TypeError("time_period must be type float/int/None.")
```

Comment:
I've too less knowledge about the type check. So I ask you:
in line 814, you check the type of the variable time_period. According the parameter definition, float, int and None are accepted. The definition of the parameter are ... f0: Optional[float] = None, time_period: Optional[float] = None,
My assumption: f0 and time_period are two independent parameter.
Your check is
if not isinstance(time_period, (float, int)) != f0 is not None:
but you do you check against f0? I would expect:
if not isinstance(time_period, (float, int)) != time_period is not None:
The same is for line 816, where you check start_time.
Maybe this is ok, but I do not understand the usage of 'f=0'